libcloudforensics/providers/azure/forensics.py

```diff
@@ -125,3 +125,66 @@ def CreateDiskCopy(
     raise RuntimeError(error_msg)
 
   return new_disk
+
+
+def StartAnalysisVm(
+    resource_group_name: str,
+    vm_name: str,
+    boot_disk_size: int,
+    ssh_public_key: str,
+    cpu_cores: int = 4,
+    memory_in_mb: int = 8192,
+    region: str = 'eastus',
+    attach_disks: Optional[List[str]] = None,
+    tags: Optional[Dict[str, str]] = None,
+    dst_profile: Optional[str] = None
+    ) -> Tuple['compute.AZVirtualMachine', bool]:
+  """Start a virtual machine for analysis purposes.
+
+  Look for an existing Azure virtual machine with name vm_name. If found,
+  this instance will be started and used as analysis VM. If not found, then a
+  new vm with that name will be created, started and returned. Note that if a
+  new vm is created, you should provide the ssh_public_key parameter.
+
+  Args:
+    resource_group_name (str): The resource group in which to create the
+        analysis vm.
+    vm_name (str): The name for the virtual machine.
+    boot_disk_size (int): The size of the analysis VM boot disk (in GB).
+    ssh_public_key (str): A SSH public key data to associate with the
+        VM. This must be provided as otherwise the VM will not be
+        accessible.
```

Comment:
Is this a path to the key? the b64 blob?

Reply:
The actual OpenSSH formatted key `ssh-rsa AAAAB3Nz...dxxEZWU=`
Added indication in the docstring